index.js

```diff
@@ -59,48 +48,62 @@ fs.readFile(file, 'utf8', (err, data) => {
            - these are in clipping[1], need to extract the relevant parts
         */
 
-        // when called this function creates a new book Object using the details above
-       function createBook (title, highlight) {
-        let book = {
-            title: title,
-            highlights: [highlight]
-        }
-        return book;
-       }
-
-        if (!books.length) {
-            //if books is empty, call createBooks and push the created Object to books
-            let book = createBook(title, highlight);
-            books.push(book);
+        if (books[title]) {
```

Comment:
you can see how much simpler the aggregation logic becomes when you use an object rather than an array.